### ISSUE TITLE
Data Augmentation erfnet-culane-pytorch

### ISSUE THREAD

**Issue by nisha1729** [cardwing/Codes-for-Lane-Detection#306]
Hi, could you explain why you are scaling the image by (0.595, 0.621) in the data augmentation side? Also, why have you written your own functions for cropping and resizing instead of pytorch's inbuilt functions? Is there some specific reason? Thanks!

```
train_loader = torch.utils.data.DataLoader(
        getattr(ds, args.dataset.replace("CULane", "VOCAug") + 'DataSet')(data_list=args.train_list, transform=torchvision.transforms.Compose([
            tf.GroupRandomScale(size=(0.595, 0.621), interpolation=(cv2.INTER_LINEAR, cv2.INTER_NEAREST)),
            tf.GroupRandomCropRatio(size=(args.img_width, args.img_height)),
            tf.GroupRandomRotation(degree=(-1, 1), interpolation=(cv2.INTER_LINEAR, cv2.INTER_NEAREST), padding=(input_mean, (ignore_label, ))),
            tf.GroupNormalize(mean=(input_mean, (0, )), std=(input_std, (1, ))),
        ])), batch_size=args.batch_size, shuffle=True, num_workers=args.workers, pin_memory=False, drop_last=True)

**Comment by nisha1729** [comment]
Could you also explain how you defined this value for ``` model.input_mean = [103.939, 116.779, 123.68].```
Will this ```input_mean``` affect the model performance on a different dataset?

**Comment by cardwing** [comment]
[103.939, 116.779, 123.68] is the mean pixel value from ImageNet. I think this value will not have a large effect on the model performance on a different benchmark. (0.595, 0.621) is just the interval the random scale ratio lies in. The rescaled height and width are slightly larger than the cropped value. I just borrow the codebase from others and hence I did not use the pytorch built-in functions.

**Comment by nisha1729** [comment]
Thank you for your reply. I'm trying to use this model for my images. Unfortunately, the model performs poorly when there is any vehicle in the images, while it performs pretty good with similar images in the CULane dataset. Do you have any suggestions on what could be the reason? My images are 6000x4000, but I'm rescaling and cropping them to the same dimensions that the network needs (976 x 208).

**Comment by cardwing** [comment]
I think maybe the domain gap in two different datasets leads to the result. You can try some domain adaptation methods to improve the performance of the model on your images. And I guess you do not have labels for your images and that's why you directly apply the trained model on CULane to your images.

**Comment by nisha1729** [comment]
Thank you. Yes, I currently do not have labels. On another note, does the network detect curved lanes?

**Comment by cardwing** [comment]
Yes. However, the performance is unsatisfactory.
